src/api/schema/events/metric.rs

```diff
@@ -0,0 +1,78 @@
+use async_graphql::Object;
+use chrono::{DateTime, Utc};
+use vector_core::event::metric::MetricTags;
+
+use super::EventEncodingType;
+use crate::{
+    config::OutputId,
+    event::{self},
+};
+
+#[derive(Debug, Clone)]
+pub struct Metric {
+    output_id: OutputId,
+    event: event::Metric,
+}
+
+impl Metric {
+    pub const fn new(output_id: OutputId, event: event::Metric) -> Self {
+        Self { output_id, event }
+    }
+}
+
+#[Object]
+/// Metric event with fields for querying metric data
+impl Metric {
+    /// Id of the component associated with the metric event
+    async fn component_id(&self) -> &str {
+        self.output_id.component.id()
+    }
+
+    /// Metric timestamp
+    async fn timestamp(&self) -> Option<&DateTime<Utc>> {
+        self.event.data().timestamp().as_ref()
+    }
+
+    /// Metric name
+    async fn name(&self) -> &str {
+        self.event.name()
+    }
+
+    /// Metric namespace
+    async fn namespace(&self) -> Option<&str> {
```

Comment:
I would have expected these fields (`namespace`, `kind`, etc.) to show up in the generated schema above, but they don't seem to be? Am I missing something? I do see the fields are actually returned by queries.

Reply:
I actually do see them in your follow-up PR, which is odd: https://github.com/vectordotdev/vector/pull/11234/files#diff-65a33a582e1355cb4649213aceb696fe92a2339b16399c84137b511f696f90f8

Reply:
Ah good catch, I think I may have missed re-running `make build-graphql-schema` at some point on this branch. Will update this.